src/main/java/com/yahoo/bullet/storage/StorageManager.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    /**
+     * Store a given ID and String value for that ID into the storage.
+     *
+     * @param id The unique ID to represent this entry.
+     * @param value The String value to store for this entry.
+     * @return A {@link CompletableFuture} that resolves to true or false if the storage was successful.
+     */
+    public CompletableFuture<Boolean> putString(String id, String value) {
+        return put(id, toBytes(value));
+    }
+
+    /**
+     * Retrieves all the IDs stored with {@link #putString(String, String)} or {@link #put(String, byte[])} from the
+     * storage as Strings.
+     *
+     * @return A {@link CompletableFuture} that resolves to a {@link Map} of IDs to their stored values as Strings or
+     *         null if no data is present.
+     */
+    public CompletableFuture<Map<String, String>> getAllString() {
+        return getAll().thenApplyAsync(StorageManager::toStringMap);
+    }
+
+    /**
+     * Retrieves a given ID from the storage stored using {@link #put(String, byte[])}.
+     *
+     * @param id The unique ID to retrieve from the storage.
+     * @return A {@link CompletableFuture} that resolves to the stored value for this ID or null if it does not exist
+     *         as a byte[]. It completes exceptionally if there were issues.
+     */
+    public abstract CompletableFuture<byte[]> get(String id);
+
+    /**
+     * Removes a given ID from the storage stored using {@link #put(String, byte[])}.
+     *
+     * @param id The unique ID to remove from the storage.
+     * @return A {@link CompletableFuture} that resolves to the stored value for this ID or null if it does not exist
+     *         as a byte[]. It completes exceptionally if there were issues.
+     */
+    public abstract CompletableFuture<byte[]> remove(String id);
+
+    /**
+     * Store a given ID and value for that ID into the storage.
+     *
+     * @param id The unique ID to represent this entry.
+     * @param value The value to store for this entry.
+     * @return A {@link CompletableFuture} that resolves to true if the storage was successful.
+     */
+    public abstract CompletableFuture<Boolean> put(String id, byte[] value);
+
+    /**
+     * Stores a map of IDs and values into the storage.
+     *
+     * @param data The map of IDs and values to store.
+     * @return A {@link CompletableFuture} that resolves to true if the storage was completely successful.
+     */
+    public abstract CompletableFuture<Boolean> putAll(Map<String, byte[]> data);
+
+    /**
+     * Retrieves all the IDs stored with {@link #putString(String, String)} or {@link #put(String, byte[])} from the
+     * storage as byte[].
+     *
+     * @return A {@link CompletableFuture} that resolves to a {@link Map} of IDs to their stored values as byte[] or
+     *         null if no data is present. It completes exceptionally if there were issues.
+     */
+    public abstract CompletableFuture<Map<String, byte[]>> getAll();
```

Comment:
Why return null when data isn't present?

Reply:
You want it to resolve to an empty map instead?

Reply:
Yea, that seems more natural in my opinion

Reply:
I didn't want to because get resolves to null if there's no data